### PR TITLE
[DOCKER] Use assembly jars for UC images

### PR DESCRIPTION
**Pull request by jaceklaskowski**
As reported in https://github.com/unitycatalog/unitycatalog/issues/339, Docker containers fail to start after a successful docker image build.

This is due to [this](https://github.com/unitycatalog/unitycatalog/commit/119f4707d033add34c0894916f544785e110fe94) and [this](https://github.com/unitycatalog/unitycatalog/commit/009ac77e7cd7926acb061ed1b8b5c3e5177c1bf7) commits that introduced assembly jars with incomplete metadata (`mainClass`).

This PR is supposed to fix https://github.com/unitycatalog/unitycatalog/issues/339 with some extra (opinionated) changes like:

1. UC CLI container accepts regular CLI options
2. No more `network` and `volume` (which may or may not be useful but turned out a nightmare to make right to achieve the above)

There are some FIXMEs left to work on in followup PRs 😉

### CODE REVIEW

**Comment by dennyglee** [comment]
Thanks @jaceklaskowski  - reviewing this now :)

File: docker/README.md [hunk 1]
```diff
@@ -1,73 +1,111 @@
...
+Start the Unity Catalog Localhost Reference Server.
+
+```shell
+./bin/start-uc-server
 ```
 
-> \[!TIP\]
-> You can run the docker container with different settings such as using a different network, volume, or ports.
-> The run script is only for demo and happy path.
+Execute `start-uc-cli-in-docker` to connect to the Unity Catalog server on your local machine (`localhost`).
```
Comment (dennyglee) [inline docker/README.md:77]: This was already done in L35

**Review by dennyglee — CHANGES_REQUESTED** [review]
Thanks @jaceklaskowski  - I really like the smoother flow of this docker readme.  Some clarifications in the `readme.md`.  Thanks!

**Comment by jaceklaskowski** [comment]
Back to the game after a short holiday...

**Comment by jaceklaskowski** [comment]
Think all the nits were addressed and the PR is ready for another review @dennyglee 🙏

File: docker/README.md [hunk 2]
```diff
@@ -1,73 +1,111 @@
...
 
 ```bash
 ./docker/bin/build-uc-server-docker
 ```
 
-This will create an image named `unitycatalog` and the version of the image would follow the version of the active git branch.
+`build-uc-server-docker` creates an image named `unitycatalog` with the version from [version.sbt](../version.sbt).
 
-#### 1.1 Unity Catalog CLI
+```console
```
[the remark this answers is not in the document]
Comment (dennyglee) [reply docker/README.md]: Thanks a bunch!

**Review by dennyglee — APPROVED** [review]
This looks great, thanks a bunch for tackling this @jaceklaskowski

**Pull request merged**